src/dune_pkg/package_universe.mli

```diff
@@ -0,0 +1,26 @@
+open! Import
+
+(** All of the packages in a dune project including local packages and packages
+    in a lockdir. The lockdir is guaranteed to contain a valid dependency
+    solution for the local packages. *)
+type t = private
+  { local_packages : Local_package.t Package_name.Map.t
+  ; lock_dir : Lock_dir.t
+  }
+
+module Error : sig
+  type t =
```

Comment:
Do we distinguish this error anywhere? If not, just making it a `User_message.t` is simpler

Reply:
I wanted to make explicit in the api the different ways that a lockdir can be invalid w.r.t local packages.

Reply:
Sure, it's already explicit in the implementation. The callers clearly don't care (at the moment), so might as well hide that detail from them.